opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Group.java

```diff
@@ -301,12 +304,12 @@ public static Group getByName(String name) {
      * the project's description.
      *
      * @param project the project
-     * @param groups set of groups
+     * @param map set of groups
      * @return set of groups matching the project
      */
-    public static Set<Group> matching(Project project, Set<Group> groups) {
-        Set<Group> copy = new TreeSet<>(groups);
-        copy.removeIf(g -> !g.match(project));
+    public static Map<String, Group> matching(Project project, Map<String, Group> map) {
+        Map<String, Group> copy = new TreeMap<>(map);
+        ((Collection<Group>) copy).removeIf(g -> !g.match(project));
```

Comment:
This will not work. `Map` is not a `Collection`:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html